test/cases/literals/heredoc-dollar-no-lead-space.hack

```diff
@@ -0,0 +1,4 @@
+// This test currently fails but should pass
+<<<EOT
+$()
+EOT;
```

Comment:
Let's add final new lines to these files... I forget why it matters (or if it actually matters, but that red stop-looking sign looks scary). You can update VSCode to automatically add a final new line after saving a file.

<img width="624" alt="Screen Shot 2021-07-01 at 12 33 54 AM" src="https://user-images.githubusercontent.com/1281153/124084602-04c9f200-da04-11eb-816b-193ba8bd5164.png">